### DISCUSSION TITLE
How can I configure deno to use a socks5 proxy?

### DISCUSSION THREAD

**Discussion by justinmchase** [denoland/deno#17606] [Q&A]
I am attempting to connect with a tls socket to a kafka server, but I need to go through a socks5 proxy to do it.

I'm trying to use `HTTPS_PROXY=localhost:50000 deno run -A index.ts`

The localhost:50000 port is actually a `kubectl port-forward ...` command to a pod running a socks5 server. I know this works because I can use it to connect to other things, I'm just trying to write a deno script to automate some stuff and have it go through that port. So I'm just curious if how I can do socks5 proxying with Deno?

**Comment by andykais** [comment]
I think what youre looking for is `Deno.HttpClient` https://deno.land/api@v1.30.0?unstable&s=Deno.HttpClient. It essentially is deno's option bag for server specific features that can be added on to the browser's fetch api. For your example, I believe something like this could work:

```ts
const client = Deno.createHttpClient({
  proxy: {
    url: Deno.env.get('HTTPS_PROXY') ?? 'socks5://127.0.0.1:5000'
  }
})
const req = await fetch('https://myserver.com', { client })
```

**Reply by andykais** [reply]
[edit] this is a more useful doc link https://deno.land/api@v1.30.0?unstable=&s=Deno.createHttpClient

**Reply by justinmchase** [reply]
Yeah but the socket is not http, its a tls connection, its own protocol. The server is kafka.

I see this library:
https://deno.land/x/socks5

But it does not implement a `connectTls` version of the proxy...

It seems like the tls connection itself should have some proxy options right on it though, is that not the case?

**Reply by ebrahimkarimi** [reply]
Im having the same issue except i get createHttpClient is not a function!

**Reply by andykais** [reply]
@ebrahimkarimi you may need to pass `--unstable` to your deno command
